pkg/controller/tektoninstallation/tektoninstallation_controller.go

```diff
@@ -87,37 +105,111 @@ func (r *ReconcileTektonInstallation) Reconcile(request reconcile.Request) (reco
+	default:
+		return reconcile.Result{}, r.statusUpdate(reqLogger, tektonInstallation, r.setStatusTektonUnknown, "tekton installation status is unknown")
 	}
-	return r.statusUpdate(logger, tektonInstallation, r.setStatusTektonSubscriptionReady, "")
 }
 
-func (r *ReconcileTektonInstallation) ensureTektonSubscription(logger logr.Logger, tektonInstallation *v1alpha1.TektonInstallation, ns string) error {
+func (r *ReconcileTektonInstallation) ensureTektonSubscription(logger logr.Logger, tektonInstallation *v1alpha1.TektonInstallation, ns string) (bool, error) {
 	sub := &olmv1alpha1.Subscription{}
 	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: ns, Name: SubscriptionName}, sub)
 	if err != nil && errors.IsNotFound(err) {
 		tektonSub := NewSubscription(ns)
 		logger.Info("Creating subscription for tekton", "Subscription.Namespace", ns, "Subscription.Name", tektonSub.Name)
 		if err := controllerutil.SetControllerReference(tektonInstallation, tektonSub, r.scheme); err != nil {
-			return err
+			return false, err
 		}
-		return r.client.Create(context.TODO(), tektonSub)
+		if err := r.client.Create(context.TODO(), tektonSub); err != nil {
+			return false, err
+		}
+		return true, nil
 	}
-	return err
+
+	return false, err
 }
 
-func (r *ReconcileTektonInstallation) setStatusTektonSubscriptionReady(tektonInstallation *v1alpha1.TektonInstallation, message string) error {
+func (r *ReconcileTektonInstallation) ensureWatchTektonConfig() (bool, error) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	if r.watchTektonConfig != nil {
+		tektonCfg := &config.Config{}
+		if err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: "default", Name: "default"}, tektonCfg); err != nil {
+			if meta.IsNoMatchError(err) {
+				log.Info("Tekton resource type does not exist yet", "message", err.Error())
+				return true, nil
+			}
+			if !errors.IsNotFound(err) { // ignore NotFound
+				log.Error(err, "Unexpected error while getting a TektonConfig to ensure a TektonConfig watcher can be created")
+				return false, err
+			}
+		}
+		if err := r.watchTektonConfig(); err != nil {
+			log.Error(err, "Unexpected error while creating a watcher on the Tekton resources", "message")
+			return false, err
+		}
+		log.Info("Added a watcher on the TektonConfig resources")
+		r.watchTektonConfig = nil // make sure watchTektonConfig() should NOT be called afterwards
+	}
+	log.Info("Watcher on the Tekton resources already added")
+	return false, nil
+}
+
+func getTektonConfigStatus(tektonCfg *config.Config) (config.InstallStatus, string) {
+	for _, conditions := range tektonCfg.Status.Conditions {
+		code := conditions.Code
+		if code == config.InstalledStatus || code == config.InstallingStatus || code == config.ErrorStatus {
+			return code, conditions.Details
+		}
+	}
+	return "unknown", ""
+}
+
+func (r *ReconcileTektonInstallation) setStatusTektonInstallationSucceeded(tektonInstallation *v1alpha1.TektonInstallation, message string) error {
```

Comment:
The message is still not used:
```suggestion
func (r *ReconcileTektonInstallation) setStatusTektonInstallationSucceeded(tektonInstallation *v1alpha1.TektonInstallation, _ string) error {
```